File: 0128_longest_consecutive_sequence/Solution.java
// runtime: 70.36%, memory: 6.09%
class Solution {
    public int longestConsecutive(int[] nums) {
        int _max = 0;
        Set<Integer> _set = new HashSet<>();
        for (int num : nums) {
            _set.add(num);
        }

        for (int num : _set) {
            if (!_set.contains(num - 1)) {
                int length = 1;

                while (_set.contains(num + length)) {
                    length++;
                }

                _max = Math.max(_max, length);
            }
        }

        return _max;
    }
}

File: 0128_longest_consecutive_sequence/solution.py
from typing import List, Set

# runtime: 74.21%, space: 66.32%
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        _set: Set[int] = set(nums)
        _max: int = 0
        
        for n in _set: 
            if n - 1 not in _set:
                l = 1

                while n + l in _set:
                    l += 1

                _max = max(_max, l)
        
        return _max
    
